File: src/backend/music-server/music-api/src/main/java/com/lalala/music/external/kafka/KafkaProducer.java
package com.lalala.music.external.kafka;

import com.lalala.event.LalalaEvent;
import lombok.RequiredArgsConstructor;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class KafkaProducer {
    private final KafkaTemplate<String, LalalaEvent> template;

    @Async
    public void execute(LalalaEvent event) {
        template.send(event.getTopic(), event);
    }
}

File: src/backend/streaming-server/src/main/kotlin/com/lalala/streaming/external/kafka/KafkaProducer.kt
package com.lalala.streaming.external.kafka

import com.lalala.event.LalalaEvent
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Component

@Component
class KafkaProducer(
    private val template: KafkaTemplate<String, LalalaEvent>
) {
    @Async
    fun execute(event: LalalaEvent) {
        template.send(event.topic, event)
    }
}
